main.py

```diff
@@ -78,6 +79,11 @@ def process_keyboard_events(actions_while_key_pressed, actions_on_key_down):
             action.execute_action()
 
 
+def process_each_frame_events(actions_each_frame):
+    for act in actions_each_frame:
+        act.execute_action()
```

Comment:
For getting a new state of an object, until now '.update' was used. Is it intentional to change this here?